### PR TITLE
PROV-91 Added tests for methods in providerManagementUtils

### PR DESCRIPTION
**Pull request by jpetersen6202**
Added test cases for getSupervisors:
Contribution for PROV-91. Added three test cases for getSupervisors;
Testing that the list returned is not empty after setting a
supervisor/supervisee relationship, Testing that the list returned is
empty on a provider with no supervisor, and that the list returned is of
proper length when a supervisor/supervisee relationship exists.

Added test cases for clearTimeComponent:
Contribution for PROV-91. Added three test cases for clearTimeComponint;
Testing the date returned when passed a day that has already started (
(hour, min, second) != (0, 0, 0) ), testing the date returned when
passed a date with a negative value in the date, and testing the date
returned when passed a day that has just started ( (hour, min, second)
== (0, 0, 0) ).

Added test cases for getSupervisees:
Contribution for PROV-91. Added three test cases for getSupervisees;
Testing that the list returned is not empty after setting a
supervisor/supervisee relationship, Testing that the list returned is
empty on a provider with no supervisee, and that the list returned is of
proper length when a supervisor/supervisee relationship exists.

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 1]
```diff
@@ -31,7 +53,137 @@
...
+   	public void shouldSetupContext() {
+   		assertNotNull(Context.getService(ProviderManagementService.class));
+   	}
+    
+    @Test
+    public void checkWhenSupervisorListIsNotEmpty() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = new Person();
+    	person.setId(2);
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: Person has a constructor so you can just do Person(2);

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 2]
```diff
@@ -31,7 +53,137 @@
...
+    	Person supervisor = new Person();
+    	supervisor.setId(2);
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: You should give the supervisor a different ID than the person

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 3]
```diff
@@ -31,7 +53,137 @@
...
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	provider.setPerson(person);
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: This should happen before the assignProviderToSupervisor line.

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 4]
```diff
@@ -31,7 +53,137 @@
...
+    	Person supervisorForPerson = providerManagementService.getSupervisorsForProvider(person).get(0);
+    	List<ProviderPersonRelationship> supervisors = new ArrayList<ProviderPersonRelationship>();
+    	Relationship relationship = null;
+    	RelationshipType relationshiptype = providerManagementService.getSupervisorRelationshipType();
+    	supervisors.add(new ProviderPersonRelationship(provider.getPerson(), provider.getIdentifier(), supervisorForPerson.getId(),  relationship, relationshiptype));
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: Are these previous 5 lines necessary?  What are they testing?

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 5]
```diff
@@ -31,7 +53,137 @@
...
+    	Assert.assertFalse(ProviderManagementUtils.getSupervisors(provider).isEmpty());
+    }
+    
+    @Test
+    public void checkWhenSupervisorListIsEmpty() throws PersonIsNotProviderException {
+    	Provider provider = new Provider();
+    	Person person = new Person();
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: Same thing here, you can just do Person(2)

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 6]
```diff
@@ -31,7 +53,137 @@
...
+    @Test
+    public void checkForTheAccurateNumberOfSupervisorsAdded() throws PersonIsNotProviderException, InvalidSupervisorException, ProviderAlreadyAssignedToSupervisorException, DateCannotBeInFutureException {
+    	Provider provider = new Provider();
+    	provider.setIdentifier("1000X");
+    	Person person = new Person();
+    	person.setId(2);
+    	Person supervisor = new Person();
+    	supervisor.setId(2);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	provider.setPerson(person);
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: This should happen before you assignProviderToSupervisor

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 7]
```diff
@@ -31,7 +53,137 @@
...
+    	Person supervisorForPerson = providerManagementService.getSupervisorsForProvider(person).get(0);
+    	List<ProviderPersonRelationship> supervisors = new ArrayList<ProviderPersonRelationship>();
+    	Relationship relationship = null;
+    	RelationshipType relationshiptype = providerManagementService.getSupervisorRelationshipType();
+    	supervisors.add(new ProviderPersonRelationship(provider.getPerson(), provider.getIdentifier(), supervisorForPerson.getId(),  relationship, relationshiptype));
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: Again, I'm confused as to what the previous five lines are supposed to be doing?

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 8]
```diff
@@ -31,7 +53,137 @@
...
+    	supervisor.setId(2);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	provider.setPerson(supervisor);
+    	Person superviseeForSupervisor = providerManagementService.getSuperviseesForSupervisor(supervisor).get(0);
+    	List<ProviderPersonRelationship> supervisees = new ArrayList<ProviderPersonRelationship>();
+    	Relationship relationship = null;
+    	RelationshipType relationshiptype = providerManagementService.getSupervisorRelationshipType();
+    	supervisees.add(new ProviderPersonRelationship(provider.getPerson(), provider.getIdentifier(), superviseeForSupervisor.getId(),  relationship, relationshiptype));
+    	Assert.assertFalse(ProviderManagementUtils.getSupervisees(provider).isEmpty());
+    }
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: Same general comments for this test as the others.

File: api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java [hunk 9]
```diff
@@ -31,7 +53,137 @@
...
+    	supervisor.setId(2);
+    	providerManagementService.assignProviderToSupervisor(person, supervisor);
+    	provider.setPerson(supervisor);
+    	Person superviseeForSupervisor = providerManagementService.getSuperviseesForSupervisor(supervisor).get(0);
+    	List<ProviderPersonRelationship> supervisees = new ArrayList<ProviderPersonRelationship>();
+    	Relationship relationship = null;
+    	RelationshipType relationshiptype = providerManagementService.getSupervisorRelationshipType();
+    	supervisees.add(new ProviderPersonRelationship(provider.getPerson(), provider.getIdentifier(), superviseeForSupervisor.getId(),  relationship, relationshiptype));
+    	Assert.assertEquals(1, ProviderManagementUtils.getSupervisees(provider).size());
+    }
```
Comment (mogoodrich) [inline api/src/test/java/org/openmrs/module/providermanagement/ProviderManagementUtilsTest.java]: Same general comments for this test as the others.

**Comment by jpetersen6202** [comment]
Thanks! I'll look into these.

**Pull request closed without merging**